### PR TITLE
Add Windows CI test runs via AppVeyor

### PR DESCRIPTION
**Pull request by pawroman**
The tests run on appveyor now! With test reports and all.

Follow up from #14 and includes initial work from @dwillmer - I suggest closing that PR.

You can see the builds on my fork here: https://ci.appveyor.com/project/pawroman/fastats

I suggest we enable AppVeyor builds for the main repo now! 🎉

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  # purpose but it is problematic because it tends to cancel builds pushed
+  # directly to master instead of just PR builds (or the converse).
+  # credits: JuliaLang developers.
+  - ps: if ($env:APPVEYOR_PULL_REQUEST_NUMBER -and $env:APPVEYOR_BUILD_NUMBER -ne ((Invoke-RestMethod `
+        https://ci.appveyor.com/api/projects/$env:APPVEYOR_ACCOUNT_NAME/$env:APPVEYOR_PROJECT_SLUG/history?recordsNumber=50).builds | `
+        Where-Object pullRequestId -eq $env:APPVEYOR_PULL_REQUEST_NUMBER)[0].buildNumber) { `
+          throw "There are newer queued builds for this pull request, failing early." }
+
+  # Prepend miniconda binaries to the %PATH%
+  - "SET PATH=%MINICONDA%\\Scripts;%PATH%"
```
Comment (pawroman) [inline .appveyor.yml:35]: Miniconda **just works** now ;)

File: .appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+build: off
+
+test_script:
+  # Run the project tests
+  - "pytest --junit-xml pytest_report.xml"
+
+after_test:
+  # Upload test report
+  # See: https://www.appveyor.com/docs/running-tests/#uploading-xml-test-results
+  - "http -f POST https://ci.appveyor.com/api/testresults/junit/%APPVEYOR_JOB_ID% pytest_report.xml@pytest_report.xml"
```
Comment (pawroman) [inline .appveyor.yml:76]: They have some PowerShell script to submit that, but it's broken - so I'm using the excellent [httpie](https://httpie.org/) utility to POST the test results: https://ci.appveyor.com/project/pawroman/fastats/build/1.0.12/job/x5tmcn4terngxugw/tests

File: .appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  - "pip install pytest"
+  - "pip install pytest-cov"
+  - "pip install codecov"
+  - "pip install sphinx"
+  - "pip install sphinx_rtd_theme"
+  - "pip install statsmodels"
+  - "pip install httpie"  # for uploading test report xml
+
+# Ignore the MSBuild nonsense
+build: off
```
Comment (pawroman) [inline .appveyor.yml:67]: I don't think we'll be shipping any compiled files for now - so no build step is needed.

**Comment by dwillmer** [comment]
Closes #14

**Comment by dwillmer** [comment]
Awesome stuff, thanks very much for taking the time for this! I was totally fed up with the windoze builds!

**Review by dwillmer — COMMENTED** [review]
Big thumbs up, thanks! 👍

**Pull request merged**